server/server.go

```diff
@@ -19,18 +20,44 @@ var (
 type listenerGenerator func(net.Listener) net.Listener
 
 type Server struct {
-	Addr net.Addr
-	Tls  bool
-
-	handler            http.Handler
```

Comment:
It makes sense to remove the handler from the attributes of the server, but did this come from the race detector or is just refactoring?